### ISSUE TITLE
Can abp Blazor be this fast?

### ISSUE THREAD

**Issue by sturlath** [abpframework/abp#13114]
Steves Sanderson Blazor WASM is.. well.. Blazor fast.. can abp.io get this fast please [695423](https://twitter.com/davidfowl/status/1540713073626865665?s=21&t=kjYAvzwMm-mzJM1WYA0VSQ)!

Sorry but abo.io’s Blazor is super slow..

**Comment by hikalkan** [comment]
ABP.IO has not its own Blazor. We are using the same one. Just adding some more libraries, adding authorization and some other real business requirements. Blazor is getting slow when we add these.

ABP's Angular UI, for example, is super fast. Try https://commercial-demo.abp.io/ for example, which is a large application.
ABP's Angular UI is using the same backend with the Blazor UI, which is also built with ABP.

I noted that. If I have time I will add ABP to [this example](https://github.com/aspnet/AspLabs/tree/main/src/QuickGrid/samples) to see how ABP is making it slower.

**Comment by sturlath** [comment]
Yes sorry for this "outburst" of mine 😅 and I know yours is not special version. To note  I´m happy with everything abp.io but loading of my Blazor backend is terribly slow (by any standard) what ever I have tried.

So when you see something like [this](https://twitter.com/stevensanderson/status/1540361092249026560) you wonder why mine is taking such a long time. 

Then you see that he is using ["It’s a combination of static pre-rendering and max trimming settings"](https://twitter.com/stevensanderson/status/1540992541641920512?s=20&t=fiBa5tPg3iYESVCuUGYOZg) and this [BlazorWasmPreRendering nuget](https://twitter.com/stevensanderson/status/1540473515245289472?s=20&t=fiBa5tPg3iYESVCuUGYOZg)

I have looked at lists like this [Blazor performance best practices](https://docs.microsoft.com/en-us/aspnet/core/blazor/performance?view=aspnetcore-6.0) and they don´t mention this package for instance.

But now that we can see that WASM can be Blazing fast so why not default file new from abp.io? Can something be done or is the framework just to heavy (doing much more than Steves version)?

**Comment by hikalkan** [comment]
OK, we will test and work on these. Thanks.

**Comment by omer-repo** [comment]
Actually, it is wonderful developing Blazor with ABP framework. It makes developing so easy to communicate UI with backend. I see only loading time problem even after caching assemblies.

![Loading after caching](https://i.imgur.com/qPfwmo4.png)

I tried to understand why Blazor is slow on loading. There are nearly 5-6 seconds gap between requests from backend. I am not sure it is related to backend side. So I imported source projects of ABP from github (v5.2.2 and v5.3.1) and checked loading times.
I couldn't find what triggers openid-configuration so I investigated "application-configuration" request. Even it takes only 90ms to download it from backend it takes 2 seconds to complete initializing module "WebAssemblyCachedApplicationConfigurationClient"

![Loading after caching](https://i.imgur.com/FjENi21.png)
![image](https://user-images.githubusercontent.com/11836388/177198786-6108f3bf-259f-4f8f-b717-eddccfa51c99.png)
![image](https://user-images.githubusercontent.com/11836388/177199712-74b228c0-e32a-4d54-8d20-eb9737a1c54c.png)
![image](https://user-images.githubusercontent.com/11836388/177197556-5551ea39-5142-4c69-b9da-6df2b6ae7969.png)


I checked also InitializeAsync of  "WebAssemblyCachedApplicationConfigurationClient", it takes 2 seconds 
`        var configurationDto = await ApplicationConfigurationAppService.GetAsync();`

As I understand from browser and log files, fetching data from backend is so fast, but processing it in Blazor side is slow. It goes to AbpApplicationConfigurationClientProxy but I couldn't investigate more because I need to import other projects to understand what happens there. 

I hope there is not a big issue while initializing modules of Blazor and it can be solved soon.

**Comment by maliming** [comment]
Thanks @omer-repo I will check it.

**Comment by sturlath** [comment]
Hi we have figured that most of our problems is related to Blazor compressed dll´s **not** served to the browser (they get created) so we are trying to add `autostart="false"`  to `<script src="_framework/blazor.webassembly.js" ></script>` that I think you are adding in [BundlingService.cs](https://github.com/abpframework/abp/blob/9271edc99c348cf37684d9a9ed871e28e946a3bd/framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/Bundling/BundlingService.cs#L117)

Here is a[ quick video ](https://www.youtube.com/watch?v=LM7-jWXIIPM) about it.

I created a [support ticked](https://support.abp.io/QA/Questions/3577/Blazor-WASM-creating-compressed-dlls-but-not-serving-them-to-the-browser-causing-login-times-of-40-50-sek) for this issue. Shouldn´t the compression be default in the template?

**Comment by maliming** [comment]
Thanks @sturlath 👍

**Comment by sturlath** [comment]
We are [having problems ](https://support.abp.io/QA/Questions/3577/Blazor-WASM-creating-compressed-dlls-but-not-serving-them-to-the-browser-causing-login-times-of-40-50-sek#answer-ab34dd90-1efd-90ca-d184-3a05c49e25eb) overriding/removing-and-re-adding the script tag. 

Just asking here in case this is something that is not currently possible but could then be introduced into version 6.0? (I can´t go live with users downloading over 100 MB website can I?)

**Comment by realLiangshiwei** [comment]
I will check it

**Comment by sturlath** [comment]
@realLiangshiwei did you by any change have time checking this out? 

Being able to override this would be great but next best would be to only have to wait for abp 6.0 and not 7.0 (thats the reason for my pestering sorry)

**Comment by realLiangshiwei** [comment]
> @realLiangshiwei did you by any change have time checking this out?
> 
> Being able to override this would be great but next best would be to only have to wait for abp 6.0 and not 7.0 (thats the reason for my pestering sorry)

See: https://github.com/abpframework/abp/issues/13771

**Comment by omer-repo** [comment]
> Thanks @omer-repo I will check it.

Hi @maliming , could you get a look the delay issue on loading

**Comment by maliming** [comment]
hi 
We will check this asap. Thanks

**Comment by sturlath** [comment]
@realLiangshiwei Im not sure this is fixed  in 6.0 https://github.com/abpframework/abp/issues/13771 

When we try to load up a Blazor WASM its still taking long time (and MB’s) to dowload. Are  there any other steps to do?

An why isn’t this a default functionality in the file new project template? Who would ever want to serve 100 MB page?

**Comment by sturlath** [comment]
@realLiangshiwei any feedback on this? Did you manage to try this out yourself and get the download size down?

Are we really doomed to be stuck with **100 MB Blazor backends**? Is anybody using it out there?

 If I wasn´t so deep into my development I would move from abp.io's version of Blazor but I can´t so I can only ask you to figure this out for us poor souls that choose it to begin with! 

So, all feedback appreciated on this one1

**Comment by hikalkan** [comment]
I removed this from 7.0-preview milestone. It doesn't mean we don't actively tracking it. However, this is not a concrete development that we will complete in a certain date range.

**Comment by realLiangshiwei** [comment]
Ahead-of-time (AOT) compilation: https://learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-7.0#ahead-of-time-aot-compilation

**Comment by sturlath** [comment]
How about [this performance increase from Blazorise](https://github.com/Megabit/Blazorise/discussions/4291) that abp.io is missing, @hikalkan? Something we can push into 7.0?

**Comment by realLiangshiwei** [comment]
Hi, @sturlath 

See: https://github.com/abpframework/abp/issues/14734

**Comment by sturlath** [comment]
Thanks @realLiangshiwei yes, I spotted this and asked [this question](https://github.com/Megabit/Blazorise/discussions/4291).

**Comment by sturlath** [comment]
I saw [this comment](https://github.com/dotnet/aspnetcore/issues/16956#issuecomment-552475370) I don´t know if this is something that can be used in abp.io but can hopefully close these loading gaps

> You can essentially put all of these inside a link header with preload to speed things up.

**Comment by sturlath** [comment]
We are now serving our Blazor WASM with static websites and CDN í Azure and that combined with Brotli compression is making huge difference.

But I just wanted to add to @omer-repo about the "timeouts" in the loads. 

We can see 5+3 sec in loading where nothing seems to be happening and was wondering what is going on there and if there is any work going on getting that down?

![image](https://user-images.githubusercontent.com/2386572/206131841-5073cc7e-2444-43cd-83fa-a7d790106e2e.png)

**Comment by sturlath** [comment]
Its not only us wondering about these things https://support.abp.io/QA/Questions/4210/Blazor-WASM-on-Azure-App---awful-loading-times

Can somebody tell us if this is being looked at? And if you have, the results? Is it fixable or will the abp wasm never be usable?

And then if you havent looked at it, why?

**Comment by dongfo** [comment]
anyone working on it?
